### ISSUE TITLE
RAM data not working

### ISSUE THREAD

**Issue by Dunuin** [monotok/zabbix-template-freenas11#4]
None of the RAM data is working. I get errors like "snmp_parse_oid(): cannot parse OID "UCD-SNMP-MIB::memCached.0"."

I use Zabbix 5.0 and MIB of FreeNAS 11.3u4.1.

**Comment by monotok** [comment]
I currently use Zabbix 4.0.3 and FreeNAS 11.1 and don't have any plans to update at the moment, only use it for home use.

FreeNAS might have changed the MIB in a newer release. You can do a SNMP walk discovery to find all the OIDs and then try updating the value in the template. You can go to Hosts - Items - Memory and edit one of the items. The OID should be changeable. 
If that is the case and you get it working for the newer release then we can create a new branch/release for 11.3.

**Comment by Dunuin** [comment]
I snmpwalked and the IDs look fine so far:

`# snmpwalk -v2c -c MyCommunity MyIP 1.3.6.1.4.1 | grep mem
UCD-SNMP-MIB::memIndex.0 = INTEGER: 0
UCD-SNMP-MIB::memErrorName.0 = STRING: swap
UCD-SNMP-MIB::memTotalSwap.0 = INTEGER: 6291072 kB
UCD-SNMP-MIB::memAvailSwap.0 = INTEGER: 6291072 kB
UCD-SNMP-MIB::memTotalReal.0 = INTEGER: 33467776 kB
UCD-SNMP-MIB::memAvailReal.0 = INTEGER: 927028 kB
UCD-SNMP-MIB::memTotalFree.0 = INTEGER: 276272 kB
UCD-SNMP-MIB::memMinimumSwap.0 = INTEGER: 16000 kB
UCD-SNMP-MIB::memShared.0 = INTEGER: 287496 kB
UCD-SNMP-MIB::memBuffer.0 = INTEGER: 0 kB
UCD-SNMP-MIB::memCached.0 = INTEGER: 4604744 kB
UCD-SNMP-MIB::memSwapError.0 = INTEGER: noError(0)
UCD-SNMP-MIB::memSwapErrorMsg.0 = STRING:
NET-SNMP-AGENT-MIB::nsModuleName."".8.1.3.6.1.4.1.2021.4.127 = STRING: memory
NET-SNMP-AGENT-MIB::nsModuleName."".9.1.3.6.1.4.1.2021.4.100.127 = STRING: memSwapError
NET-SNMP-AGENT-MIB::nsModuleName."".9.1.3.6.1.4.1.2021.4.101.127 = STRING: memSwapErrMsg`

Zabbix shows me for example the error "snmp_parse_oid(): cannot parse OID "UCD-SNMP-MIB::memAvailReal.0"." and your template looks for "OID: .1.3.6.1.4.1.2021.4.6" / "UCD-SNMP-MIB::memAvailReal.0" and the snmpwalk returned "UCD-SNMP-MIB::memAvailReal.0 = INTEGER: 927028 kB".

I checked the FreeNAS "UCD-SNMP-MIB.txt" and the zabbix host "UCD-SNMP-MIB.txt" and they are identical.
Any idea what else could be a problem? The zfs and zpool data is working.

**Comment by monotok** [comment]
Tbh its been years since I created this so struggling to think what could be an issue. What server platform are you running this on? Ubuntu/Centos? I had to disable CentOS selinux for zabbix to work.

I would try going into the settings of the host, find the memory item and replace the OID text from the template with '.1.3.6.1.4.1.2021.4.6' and see if it works. I would guess if it does then maybe zabbix can't read the MIB file.
